Horizontal scrollbar

kpherwan opened the pull request:
Issue - https://github.com/informatics-isi-edu/chaise/issues/1573

```
/**
         * Some of the tables can be very long and the horizontal scroll only sits at the very bottom by default
         * A fixed horizontal scroll is added here that sticks to the top as we scroll vertically and horizontally
         * Scenarios -
         * 1. Record
         * 2. Recordset
         * 3. Record facer
*/
```

Test link - https://dev.rebuildingakidney.org/~kpherwan/chaise/recordset/#2/Protocol:Protocol@sort(RMT::desc::,RID)

RFSH commented:
Per our previous conversation, the previous condition wasn't ideal,

- It was waiting for the initial load and if the initial load was slow, scrollbar would show up and wasn't doing anything.
- Since we're not doing any data request in the result page of multi-create/edit, the scrollbar event wasn't getting attached.

So I changed the condition to be similar to what we have in the template. To make sure we're attaching the event handler as soon as the directive DOM is ready.

Please make sure you've properly tested this in different tables and if it works, we should be able to merge.

P.S. I don't know why a lot test cases were failing in your other commits, let's wait for the latest build and see the result. If a lot test cases failed we have to investigate.

kpherwan commented:
> Per our previous conversation, the previous condition wasn't ideal,
> 
> * It was waiting for the initial load and if the initial load was slow, scrollbar would show up and wasn't doing anything.
> * Since we're not doing any data request in the result page of multi-create/edit, the scrollbar event wasn't getting attached.
> 
> ~So I changed the condition to be similar to what we have in the template. To make sure we're attaching the event handler as soon as the directive DOM is ready.~
> 
> We actually don't need to have any condition and we can just attach the event handler in the body of `registerTableCallbacks` function. I added a comment in the code to explain this, and also refactored the code a bit.
> 
> Please make sure you've properly tested this in different tables and if it works, we should be able to merge. Also is there any way that we can make the scrolling smoother?



Sorry I accidentally closed the PR, reopened it now.

The refactor made is working perfectly, thank you so much. As discussed, I have made few changes to make the scrolling smoother and tested it on Chrome, Firefox, Safari. Let me know if it still persists.

RFSH commented on common/styles/scss/app.scss:
```diff
@@ -723,3 +723,20 @@ html {
     }
 }
+
+.top-scroll-wrapper {
```
Given that app.css is included in the static pages that have navbar, we have to make sure our rules are properly scoped and also use specific names if we can.

In this case, let's move these two new rules that you added to be in inside the `.chaise-body` wrapper above. Also let's just change the class names that you added to have `chaise-` and also more specific to table, so

- `top-scroll-wrapper` -> `chaise-table-top-scroll-wrapper`
- `top-scroll` -> `chaise-table-top-scroll`

kpherwan replied:
Makes sense, made the necessary changes
